### ISSUE TITLE
Experiencing upstream connections errors.

### ISSUE THREAD

**Issue by marion30304** [engintron/engintron#79]
Hello,
I installed engintron on a vanilla centos 6.7 with the recommended engintron.dtd. The only thing extra was mod pagespeed for apache. 
I am receiving the following errors:

"http://123.249.24.233/POST_ip_port.phpAccept: */*"
2016/02/16 17:56:53 [error] 17024#0: *5918 connect() failed (110: Connection timed out) while connecting to upstream, client: 61.160.213.56, server: localhost, request: "GET http://zc.qq.com/cgi-bin/chs/numreg/init? HTTP/1.0", upstream: "http://23.15.9.171:8080/cgi-bin/chs/numreg/init?", host: "zc.qq.com"
2016/02/16 17:56:58 [error] 17024#0: *5921 connect() failed (110: Connection timed out) while connecting to upstream, client: 61.160.213.56, server: localhost, request: "GET http://zc.qq.com/cgi-bin/chs/numreg/init? HTTP/1.0", upstream: "http://23.15.9.178:8080/cgi-bin/chs/numreg/init?", host: "zc.qq.com"
2016/02/16 18:11:59 [error] 17024#0: *5971 connect() failed (110: Connection timed out) while connecting to upstream, client: 61.160.213.247, server: localhost, request: "GET http://zc.qq.com/cgi-bin/chs/numreg/init? HTTP/1.0", upstream: "http://23.15.9.171:8080/cgi-bin/chs/numreg/init?", host: "zc.qq.com"
2016/02/16 18:12:04 [error] 17024#0: *5972 connect() failed (110: Connection timed out) while connecting to upstream, client: 61.160.213.247, server: localhost, request: "GET http://zc.qq.com/cgi-bin/chs/numreg/init? HTTP/1.0", upstream: "http://23.15.9.171:8080/cgi-bin/chs/numreg/init?", host: "zc.qq.com"
2016/02/16 18:26:13 [error] 17024#0: *6070 connect() failed (110: Connection timed out) while connecting to upstream, client: 61.160.213.55, server: localhost, request: "GET http://zc.qq.com/cgi-bin/chs/numreg/init? HTTP/1.0", upstream: "http://23.15.9.178:8080/cgi-bin/chs/numreg/init?", host: "zc.qq.com"

I have not installed csf on my setup yet and not seeing much under apache error logs. Could you please help point me in a direction that might be the cause?

Regards.

**Comment by fevangelou** [comment]
Assuming you're using Engintron v1.6.0, try raising the timeout values in nginx.conf (via WHM > Plugins > Engintron) from 20s to 30s. The error you get is because the backend (Apache) does not respond fast to Nginx's requests and there's a 20s time limit by default to wait for such a response.

The 2 values you need to change are:

    client_body_timeout			20s; # Use 5s for high-traffic sites
    client_header_timeout			20s; # Use 5s for high-traffic sites

Just change 20s to 30s and restart Nginx. Let me know if the errors persist.

By the way, you'll notice that Apache is slow if you directly call your URLs via the 8080 port. What Nginx currently serves/shows is the cached output from Apache.

**Comment by marion30304** [comment]
Thanks, I'll give it a try

**Comment by InteractM** [comment]
Same here - increased to 30s then to 60s - no luck.

**Comment by fevangelou** [comment]
If you directly call any of your URLs using port 8080 (=Apache) how long
does it take for any given URL to load? In other words, is your backend
slow too?

**Comment by InteractM** [comment]
Apache response is fast. Strange is when I will disable Engintron it still show up that NGNIX is serving the site via Wappalizer.

**Comment by fevangelou** [comment]
The only reliable way to check which server is on is by examining response
headers for one of your sites.

There are services to check that or use curl if you know how. Otherwise
Google "web curl".

Wappalyser shows you a cached image obviously if Engintron is disabled as
you say.

As for Apache responding fast and Nginx throwing upstream errors, that's
not possible.

Provide more so I can help you out if you want. So far you haven't provided
anything, be it error logs, system specs or current server configuration.

**Comment by marion30304** [comment]
Sorry for the late reply. I made the changes suggested to 30s. I am consistently receiving these errors now:

61.147.70.111, server: localhost, request: "POST http://123.249.24.233/POST_ip_port.php HTTP/1.1", upstream: "http://123.249.24.233:8080/POST_ip_port.php", host: "123.249.24.233", referrer: "http://123.249.24.233/POST_ip_port.phpAccept: */*"
2016/02/18 18:05:18 [error] 32426#0: *8382 connect() failed (110: Connection timed out) while connecting to upstream, client: 61.160.213.56, server: localhost, request: "GET http://zc.qq.com/cgi-bin/chs/numreg/init? HTTP/1.0", upstream: "http://23.205.169.130:8080/cgi-bin/chs/numreg/init?", host: "zc.qq.com"
2016/02/18 18:05:23 [error] 32426#0: *8383 connect() failed (110: Connection timed out) while connecting to upstream, client: 61.160.213.56, server: localhost, request: "GET http://zc.qq.com/cgi-bin/chs/numreg/init? HTTP/1.0", upstream: "http://23.205.169.112:8080/cgi-bin/chs/numreg/init?", host: "zc.qq.com"
2016/02/18 18:05:32 [error] 32426#0: *8390 connect() failed (111: Connection refused) while connecting to upstream, client: 222.186.30.211, server: localhost, request: "POST http://123.249.24.233/POST_ip_port.php HTTP/1.0", upstream: "http://123.249.24.233:8080/POST_ip_port.php", host: "123.249.24.233", referrer: "http://123.249.24.233/POST_ip_port.phpAccept: */*"

Additional info:
My server has a main ip along with two additional dedicated ips for my cpanel accounts. I have not modified anything regarding the module except those suggestion on the after engintron install with the exception of the mysql recommendation which will not allow me to restart mysql when I apply the changes. 

Apache error logs:

[Thu Feb 18 17:54:29.021860 2016] [mpm_prefork:warn] [pid 11883] AH00167: long lost child came home! (pid 22230)
[Thu Feb 18 17:57:26.918775 2016] [access_compat:error] [pid 25233] [client 199.175.51.29:58992] AH01797: client denied by server configuration: /home/xxxx/public_html/skin/frontend/rwd/default/scss/, referer: http://xxxx/outdoor/filter/fisher-research-labs
[Thu Feb 18 18:02:27.573870 2016] [access_compat:error] [pid 25230] [client 199.175.51.29:58909] AH01797: client denied by server configuration: /home/xxxx/public_html/skin/frontend/rwd/default/scss/, referer: http://xxxx/gps-handheld-outdoor
[Thu Feb 18 18:02:59.899872 2016] [cgi:error] [pid 25232] [client 61.160.213.56:4970] AH02811: script not found or unable to stat: /usr/local/apache/cgi-bin/chs
[Thu Feb 18 18:07:28.159486 2016] [access_compat:error] [pid 25231] [client 199.175.51.29:59249] AH01797: client denied by server configuration: /home/xxxx/public_html/skin/frontend/rwd/default/scss/, referer: http://xxxx/outdoor/filter/ronstan
[Thu Feb 18 18:07:38.284924 2016] [access_compat:error] [pid 25234] [client 199.175.51.29:60296] AH01797: client denied by server configuration: /home/xxxx/public_html/app/etc/local.xml
[Thu Feb 18 18:12:28.325610 2016] [access_compat:error] [pid 25233] [client 199.175.51.29:58992] AH01797: client denied by server configuration: /home/xxxx/public_html/skin/frontend/rwd/default/scss/, referer: http://xxxx/garmin-handlebar-mounting-bracket.html
[Thu Feb 18 18:17:21.864904 2016] [cgi:error] [pid 25232] [client 61.160.213.32:2249] AH02811: script not found or unable to stat: /usr/local/apache/cgi-bin/chs
[Thu Feb 18 18:17:29.751536 2016] [access_compat:error] [pid 25230] [client 199.175.51.29:57660] AH01797: client denied by server configuration: /home/xxxx/public_html/skin/frontend/rwd/default/scss/, referer: http://xxxx/communication/filter/iridium
[Thu Feb 18 18:22:31.572513 2016] [access_compat:error] [pid 25234] [client 199.175.51.29:57154] AH01797: client denied by server configuration: /home/xxxx/public_html/skin/frontend/rwd/default/scss/, referer: http://xxxx/communication/filter/wave-wifi


I have just done a site move from shared to a vps so some of these errors are probably related to magento.

If you need additional info please let me know and Ill supply.

Thanks

**Comment by fevangelou** [comment]
Let's clear something out first. Do any websites work on your server after you installed Engintron?

**Comment by InteractM** [comment]
On mine when Engintron is enabled - not unless used HTTPS connections. When is disabled - yes.

**Comment by fevangelou** [comment]
Can you rephrase?

**Comment by InteractM** [comment]
Engintron is enabled - going to http://www.childrensmedgroup.com/ - site is not working, NGNIX logs shows  experiencing upstream connections errors. 
Engintron is enabled - going to https://www.childrensmedgroup.com/ - site is working fine as bypass NGNIX.
Engintron is disabled - going to http://www.childrensmedgroup.com/ - site is working fine.

**Comment by fevangelou** [comment]
Is port 8080 open in your firewall?

**Comment by InteractM** [comment]
Yes - 8080 and 8081 as NgnixCP is using that one

**Comment by marion30304** [comment]
Ok, I looked into the errors with IPs. Apparently these are chinese ips that are related to scanning attacks. All my sites work. Still no sure why the error or how it relates to port 8080. I think I will keep an eye on it and see what plays out. Pingdom tools is listing in the headers as Nginx and engintron so probably all good. Unless you have some other insight I will call this resloved.
Thanks for your time.
